controllers/userController.js

```diff
@@ -0,0 +1,151 @@
+const express = require("express");
+const User = require("../model/UserModel");
+const asyncHandler = require("express-async-handler");
+const bcyrpt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+// creating a register endpoint
+
+const registerUser = asyncHandler(async (req, res) => {
+  const { username, email, password } = req.body;
+  if (!username || !email || !password)
+    res.status(400).json({ error: "all fields are mandatory" });
+
+  //checking if user already exists
+
+  const availableUser = await User.findOne({ email });
+  if (availableUser) res.status(400).json({ error: "user already exists" });
+
+  //hashed password
+
+  const hashedPassword = await bcyrpt.hash(password, 10);
+  console.log(hashedPassword);
+  const user = await User.create({
+    username,
+    email,
+    password: hashedPassword,
+  });
+  console.log(user);
+  if (user) res.status(200).json({ _id: user.id, email: user.email });
+  else {
+    res.status(400);
+  }
+  throw new Error("user data not valid");
+});
+
+//Creating a login endpoint
+const userLogin = asyncHandler(async (req, res) => {
+  const { email, password } = req.body;
+  if (!email || !password)
+    res.status(400).json({ error: "all fields are mandatory" });
+  const user = User.findOne({ email });
+
+  //comapring password with hashed password
+  if (user && (await bcyrpt.compare(password, user.password))) {
+    const accessToken = jwt.sign(
+      {
+        user: {
```

Comment:
to create an accessToken, you need just the user id, not all the user information.